lib/commits.js

```diff
@@ -76,20 +77,25 @@ module.exports.findCommitsWithAssociatedPullRequests = async ({
   const dataPath = ['repository', 'ref', 'target', 'history']
   const repoNameWithOwner = `${owner}/${repo}`
 
-  let data
+  let data, commits
   if (lastRelease) {
     log({
       app,
       context,
-      message: `Fetching all commits for branch ${branch} since ${lastRelease.published_at}`,
+      message: `Fetching all commits for branch ${branch} since ${lastRelease.created_at}`,
     })
 
     data = await paginate(
       context.github.graphql,
       module.exports.findCommitsWithAssociatedPullRequestsQuery,
-      { ...variables, since: lastRelease.published_at },
+      { ...variables, since: lastRelease.created_at },
       dataPath
     )
+    // GraphQL call is inclusive of commits from the specified dates.  This means the final
+    // commit from the last tag is included, so we remove this here.
+    commits = _.get(data, [...dataPath, 'nodes']).filter(
+      (commit) => commit.committedDate != lastRelease.created_at
```

Comment:
Should be greater than?

Reply:
I get it now. The tagged commit is the thing you want to filter out 😅

Reply:
My only hesitation here is that comparing dates could cause issues in the future. We could filter out more than just the first PR. 

For example, two merges made in the same second unlikely to happen with the GUI but could definitely happen if branches are merged locally using a script and pushed. 

Does this DateTime include milliseconds?

The alternate way is more complicated and would require a second rest API call in `releases.js` to retrieve the reference of the latest release tag. Then one could compare the commit hashes. Something like:
```node
commits = _.get(data, [...dataPath, 'nodes']).filter(
      (commit) => commit.oid != lastRelease.tag_ref
```

Reply:
Honestly I like the tag ref filter more

Reply:
It is indeed milliseconds so the chance of collision is rare. Reference is possible, but as you mention would require another API call (commit hash is not available in listReleases either by REST or GraphQL) . It is not possible to do this with 2 api calls as we have currently using commit hash. We would end up with a similar solution to #409. My view is that hash is more elegant at the cost of an additional API call. This current solution is more pragmatic and still a clear improvement over what exists on master.

Open to suggestions around way forward.

Reply:
Hmm would prefer to avoid the extra API call over very rare case of exact milliseconds 😅

Reply:
Just realised I made a mistake, format is timestamp is:
`2019-04-27T13:38:46Z`
So I suppose it is seconds not milliseconds.

Reply:
Even with seconds I would still argue we would need to save the API call.